geonode/br/management/commands/backup.py

```diff
@@ -276,80 +276,104 @@ def create_geoserver_backup(self, settings, target_folder):
         geoserver_bk_file = os.path.join(target_folder, 'geoserver_catalog.zip')
 
         print("Dumping 'GeoServer Catalog ["+url+"]' into '"+geoserver_bk_file+"'.")
-        data = {'backup': {'archiveFile': geoserver_bk_file, 'overwrite': 'true',
-                           'options': {'option': ['BK_CLEANUP_TEMP=true']}}}
-        headers = {
-            'Accept': 'application/json',
-            'Content-type': 'application/json'
-        }
-        r = requests.post(url + 'rest/br/backup/', data=json.dumps(data),
-                          headers=headers, auth=HTTPBasicAuth(user, passwd))
+        r = requests.put(url + 'rest/reset/',
+                         auth=HTTPBasicAuth(user, passwd))
+        if r.status_code != 200:
+            if (r.status_code != 200):
+                raise ValueError('Could not reset GeoServer catalog!')
+        r = requests.put(url + 'rest/reload/',
+                         auth=HTTPBasicAuth(user, passwd))
+        if r.status_code != 200:
+            if (r.status_code != 200):
+                raise ValueError('Could not reload GeoServer catalog!')
+
+        self.run_geoserver_backup(url, user, passwd, geoserver_bk_file)
+
+    def run_geoserver_backup(self, url, user, passwd, geoserver_bk_file, retries=3):
+
         error_backup = 'Could not successfully backup GeoServer ' + \
                        'catalog [{}rest/br/backup/]: {} - {}'
-        if r.status_code in (200, 201, 406):
-            try:
-                r = requests.get(url + 'rest/br/backup.json',
-                                 headers=headers,
-                                 auth=HTTPBasicAuth(user, passwd),
-                                 timeout=10)
-                if (r.status_code == 200):
-                    gs_backup = r.json()
-                    _url = gs_backup['backups']['backup'][len(gs_backup['backups']['backup']) - 1]['href']
-                    r = requests.get(_url,
+
+        _status_code = None
+        _status_text = None
+        for _r in range(retries):
```

Comment:
please @afabiani add a line of comment to explain the requirement for this cycle